hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieTableServiceClient.java

```diff
@@ -798,7 +795,11 @@ protected void archive(HoodieTable table) {
     }
     try {
       final Timer.Context timerContext = metrics.getArchiveCtx();
-      // We cannot have unbounded commit files. Archive commits if we have to archive
+      // We cannot have unbounded commit files. Archive commits if we have to archive.
+
+      // Reload table timeline to reflect the latest commits,
+      // there are some table services (for e.g, the cleaning) that executed right before the archiving.
+      table.getMetaClient().reloadActiveTimeline();
```

Comment:
Is this going to archive more instants since the view is refreshed?  Maybe we can avoid the overhead if that’s the case, since the next write should get the timeline refreshed.